### PR TITLE
Add switch to turn collectors on/off

### PR DESCRIPTION
**Pull request by dmonzonis**
This commit implements a way to turn off each collector using an environment variable. Turning it off means it won't be caught by celery, and therefore it won't be run automatically, although the collectors can still be run manually from a shell even if they have been turned off.

There is an environment variable for each of the collector groups: Bugzilla, Jira, Errata Tool, NVD and OSV. This means they can each be turned on/off individually.

The collectors are turned on by default.

Closes OSIDB-2884.

### CODE REVIEW

File: collectors/framework/models.py [hunk 1]
```diff
@@ -481,13 +482,20 @@ def collector(
                   depends on, may be left None
     dry_run     - determines whether the collector saves the
                   data or just logs them, may be left None
+    emabled:    - whether the collector is to be executed as a celery
```
Comment (MrMarble) [inline collectors/framework/models.py:485]: small typo
```suggestion
    enabled:    - whether the collector is to be executed as a celery
```

**Review by JakubFrejlach — APPROVED** [review]
Small consideration, otherwise LGTM!

File: docker-compose.yml [hunk 2]
```diff
@@ -151,6 +151,11 @@ services:
         PRODUCT_DEF_URL: ${PRODUCT_DEF_URL}
         PRODUCT_DEF_BRANCH: ${PRODUCT_DEF_BRANCH}
         PS_CONSTANTS_URL: ${PS_CONSTANTS_URL}
+        BZ_COLLECTOR_ENABLED: ${BZ_COLLECTOR_ENABLED}
+        ERRATA_COLLECTOR_ENABLED: ${ERRATA_COLLECTOR_ENABLED}
+        JIRA_COLLECTOR_ENABLED: ${JIRA_COLLECTOR_ENABLED}
+        NVD_COLLECTOR_ENABLED: ${NVD_COLLECTOR_ENABLED}
+        OSV_COLLECTOR_ENABLED: ${OSV_COLLECTOR_ENABLED}
```
[the remark this answers is not in the document]
Comment (jobselko) [reply docker-compose.yml:158]: And maybe even into `docker-compose.test.yml` to disable collectors on localhost?

File: collectors/framework/models.py (hunk 1 above)
Comment (dmonzonis) [reply collectors/framework/models.py:485]: oops

File: docker-compose.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (dmonzonis) [reply docker-compose.yml:158]: @jobselko AFAIK `docker-compose.test.yml` only defines the test pods, and the celery pods are defined in `docker-compose.yml` so adding the env vars there should be enough to be able to enable/disable them in localhost

File: collectors/bzimport/constants.py [hunk 3]
```diff
@@ -26,3 +26,5 @@
 
 # Maximum age of connection to Bugzilla in seconds, recommended value is 60
 BZ_MAX_CONNECTION_AGE = get_env("BZ_MAX_CONNECTION_AGE")
+
+COLLECTOR_ENABLED = get_env("BZ_COLLECTOR_ENABLED", default="True", is_bool=True)
```
Comment (osoukup) [inline collectors/bzimport/constants.py:30]: Could we make this on a higher level of granularity? Currently eg. on the UAT we would need the flaw collector disabled by the tracker and the metadata one enabled which is not possible with this simple approach.

File: collectors/jiraffe/tasks.py [hunk 4]
```diff
@@ -8,6 +8,7 @@
 from osidb.models import Tracker
 
 from .collectors import JiraTrackerCollector, MetadataCollector
+from .constants import COLLECTOR_ENABLED
```
Comment (osoukup) [inline collectors/jiraffe/tasks.py:11]: Here we might also get use of the switch split in some scenarios.

File: docker-compose.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (osoukup) [reply docker-compose.yml:158]: I am also thinking about the OPS repo. As most of the case we want the collectors on and they are on by default we do not have to add them there. The only case I would be for adding is to disable the UAT flaw collector. When it is enable but the BBSync is disabled it results in weird situations when something may be created internally in the DB and then synced from Bugzilla once more (eg. Assembler collector duplicates).

**Review by osoukup — CHANGES_REQUESTED** [review]
I think we need a higher granularity - on collector level, not module level. Otherwise the mechanism looks great, this is exactly what we need.

File: docker-compose.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (dmonzonis) [reply docker-compose.yml]: @osoukup I made a [PR](https://gitlab.corp.redhat.com/product-security/dev/osidb-ops/-/merge_requests/88) to add these variables in the ops repo

**Review by osoukup — APPROVED** [review]
LGTM

**Pull request merged**